src/components/Nav.astro

```diff
@@ -23,6 +23,15 @@ const currentPath = pathname;
 <script>
   const burger = document.querySelector('button.nav-toggle');
   const navList = document.querySelector('.nav');
+  const navLinks = navList ? navList.querySelectorAll('li') : [];
+  const currentURL = window.location.href;
+  navLinks.forEach((li) => {
+    const link = li.querySelector('a');
+    if (!link) return;
+    if (link.href === currentURL) {
```

Comment:
Same here. You can get rid of the previous "if" block if you use optional chaining on the link object.